### PR TITLE
Alerting: Add support for alert rule groups

### PR DESCRIPTION
**Pull request by alexweav**
Adds support for rule groups in Unified Alerting.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -77,3 +77,5 @@ require (
 	google.golang.org/grpc v1.48.0 // indirect
 	google.golang.org/protobuf v1.28.0 // indirect
 )
+
+replace github.com/grafana/grafana-api-golang-client v0.9.1 => ../grafana-api-golang-client
```
Comment (alexweav) [inline go.mod]: Depends on: https://github.com/grafana/grafana-api-golang-client/pull/103

This PR is in draft until the above is submitted and a release is made.

File: examples/resources/grafana_alert_rule/resource.tf [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        }
+        data {
+            ref_id = "B"
+            query_type = ""
+            relative_time_range {
+                from = 0
+                to = 0
+            }
+            datasource_uid = "-100"
+            model = <<EOT
```
Comment (alexweav) [inline examples/resources/grafana_alert_rule/resource.tf]: I designed this example to show 2 workflows. FIrst stage shows the HCL-based rule data, where the second stage shows how to do JSON-based rule data. Both are supported. First looks cleaner, second is nice if a user already has rule JSON ready to go.

File: examples/resources/grafana_alert_rule/_acc_multi_rule_group.tf [hunk 3]
```diff
@@ -0,0 +1,161 @@
+resource "grafana_folder" "compound_rule_folder" {
+    title = "My Compound Alert Rule Folder"
+}
+
+resource "grafana_alert_rule" "my_multi_alert_group" {
+    name = "My Multi-Alert Rule Group"
+    folder_uid = grafana_folder.compound_rule_folder.uid
+    interval_seconds = 240
+    org_id = 1
+    rules {
```
Comment (JohnnyQQQQ) [inline examples/resources/grafana_alert_rule/_acc_multi_rule_group.tf]: Nit: should we call this `rule` instead of `rules`?

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM, one small optional nit.

File: examples/resources/grafana_alert_rule/_acc_multi_rule_group.tf (hunk 3 above)
Comment (alexweav) [reply examples/resources/grafana_alert_rule/_acc_multi_rule_group.tf]: Good idea, done

**Pull request merged**